samples/excel/30-range/conditional-formatting-basic.yaml

```diff
@@ -0,0 +1,390 @@
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                const range = sheet.getRange("B8:E13");
+                const conditionalFormat = range.conditionalFormats.add(Excel.ConditionalFormatType.custom);
+                conditionalFormat.custom.rule.formula = '=IF(B8>INDIRECT("RC[-1]",0),TRUE)';
+                conditionalFormat.custom.format.font.color = "green";
+
+                await context.sync();
+            });
+        }
+
+        async function listConditionalFormats() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                sheet.load("name");
+                const worksheetRange = sheet.getRange();
+                worksheetRange.load("conditionalFormats/type");
+
+                await context.sync();
+
+                $("#conditional-formats li").remove();
+                if (worksheetRange.conditionalFormats.items.length > 0) {
+                    for (let i = 0; i < worksheetRange.conditionalFormats.items.length; i++) {
+                        let $a = $(`<a href=''>${worksheetRange.conditionalFormats.items[i].type}</a>`)
+                            .click(
+                                async (e: any) => {
+                                    e.preventDefault();
+                                    const sheet = context.workbook.worksheets
+                                        .getItem(e.data.worksheet);
+                                    const conditionalFormat = sheet.getRange()
+                                        .conditionalFormats.getItemAt(e.data.index);
+                                    const range = conditionalFormat.getRange();
+                                    range.select();
+
+                                    await context.sync();
+                            });
+                        let $li = $(`<li></li>`);
+                        $li.append($a);
+                        $("#conditional-formats").append($li);
+                        $(".conditional-formats").show();
+                    }
+                } else {
+                    showNotification("No conditional formats in workbook");
+                }
+            });
+        }
+
+        async function clearAllConditionalFormats() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                const range = sheet.getRange();
+                range.conditionalFormats.clearAll();
+                
+                await context.sync();
+
+                $(".conditional-formats").hide();
+            });
+        }
+
+        async function setup() {
+            await Excel.run(async (context) => {
+                const sheet = await OfficeHelpers.ExcelUtilities
+                    .forceCreateSheet(context.workbook, "Sample");
+                queueCommandsToCreateTemperatureTable(sheet);
```

Comment:
I'm fine with dropping the `queueCommandsTo` and just doing `createTemperatureTable`

Reply:
I've grown fond of it. :-)